wgengine/tsdns/neterr_darwin.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tsdns
+
+import (
+	"errors"
+	"syscall"
+)
+
+func isSyscallErrno(err error, errno syscall.Errno) bool {
+	var n syscall.Errno
+	if !errors.As(err, &n) {
+		return false
+	}
+	return n == errno
+}
+
+func networkIsDown(err error) bool {
+	return isSyscallErrno(err, syscall.ENETDOWN)
```

Comment:
why not errors.Is? you can prevent the allocation into the target interface if you do it once in a global variable:

```go
  var errSyscallENETDONE error = syscall.ENETDOWN
```

?

And below.

Reply:
Nice, thanks.